src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -386,50 +390,61 @@ public void buttonDeleteCvrFileClicked() {
 
   public void changeCvrFilePath(CellEditEvent cellEditEvent) {
     CVRSource cvrSelected = tableViewCvrFiles.getSelectionModel().getSelectedItem();
-    String cvrFilePath = cellEditEvent.getNewValue().toString().trim();
-    if (cvrFilePath.isBlank()) {
-      Logger.log(Level.WARNING, "Cast vote record file path is required!");
-    } else {
-      cvrSelected.setFilePath(cvrFilePath);
+    String oldFilePath = cvrSelected.getFilePath();
+    cvrSelected.setFilePath(cellEditEvent.getNewValue().toString().trim());
+    if (!cvrSourceHasRequiredFields(cvrSelected)) {
+      cvrSelected.setFilePath(oldFilePath);
```

Comment:
Could you add a comment somewhere explaining this and the similar logic below? The UI is rejecting any editing of the CVR source that creates an invalid state?

Reply:
Done.
`// Cache the original value to revert back to in case the change fails`